src/Path.Drag.js

```diff
@@ -171,18 +183,31 @@ L.Handler.PathDrag = L.Handler.extend( /** @lends  L.Path.Drag.prototype */ {
     var diff = transformation.untransform(px, scale)
       .subtract(transformation.untransform(L.point(0, 0), scale));
 
-    // console.time('transform');
+    return projection.unproject(projection.project(point)._add(diff));
+  },
+
+  /**
+   * Applies transformation, does it in one sweep for performance,
+   * so don't be surprised about the code repetition.
+   *
+   * [ x ]   [ a  b  tx ] [ x ]   [ a * x + b * y + tx ]
+   * [ y ] = [ c  d  ty ] [ y ] = [ c * x + d * y + ty ]
+   *
+   * @param {Array.<Number>} matrix
+   */
+  _transformPoints: function(matrix) {
+    var path = this._path;
+    var i, len;
+
+    var px = L.point(matrix[4], matrix[5]);
 
     // all shifts are in-place
     if (path._point) { // L.Circle
-      path._latlng = projection.unproject(
-        projection.project(path._latlng)._add(diff));
+      path._latlng = this._transformPoint(matrix, path._latlng);
       path._point._add(px);
     } else if (path._originalPoints) { // everything else
       for (i = 0, len = path._originalPoints.length; i < len; i++) {
-        latlng = path._latlngs[i];
-        path._latlngs[i] = projection
-          .unproject(projection.project(latlng)._add(diff));
+        path._latlngs[i] = this._transformPoint(matrix, path._latlngs[i]);
```

Comment:
49ms vs 22 on my machine. Small numbers but you can see the impact. that might be the method call as well